app/Controllers/CategoryController.php

```diff
@@ -0,0 +1,102 @@
+<?php
+declare(strict_types=1);
+namespace App\Controllers;
+
+
+
+use App\Model\CategoryRepository;
+
+class CategoryController extends Controller
+{
+
+    private CategoryRepository $categoryRepository;
+    public function __construct(){
+        $this->categoryRepository = new CategoryRepository();
+    }
+    public function index(): void
+    {
+        $categories = $this->categoryRepository->getAllCategories();
+
+        require_once __DIR__ . "/../../views/categories/showCategory.tmpl.php";
+    }
+
+    public function create(): void
+    {
+        $alertMessage = "";
+        $categoryName = $_POST['category_name'] ?? null;
+
+        if($_SERVER["REQUEST_METHOD"] == "POST"){
+            if (empty($categoryName)) {
+                header("HTTP/1.1 400 Bad Request");
+                exit();
+            }
+            $result = $this->categoryRepository->addCategory($categoryName);
+            $alertMessage = ($result) ? "<div class='alert alert-success' role='alert'>Category added successfully!</div>":
+                "<div class='alert alert-danger' role='alert'>Something went wrong!</div>";
+
+        }
+        $this->render("categories/addCategory", [
+            "categoryName" => $categoryName,
+            "alertMessage" => $alertMessage
+        ]);
+
+    }
+
+    public function remove(): void
+    {
+        $alertMessage = "";
+        $categoryId = (int)$_POST['category_id'] ?? null;
+        if($_SERVER["REQUEST_METHOD"] == "POST"){
+
+            $deleted = $this->categoryRepository->deleteCategory($categoryId);
+
+            if($deleted){
+                header("location: /category/show");
+            } else {
+                $alertMessage = "<div class='alert alert-danger' role='alert'>Something went wrong!</div>";
+            }
+        }
+        $this->render("categories/showCategory", [
+            "categoryId" => $categoryId,
+            "alertMessage" => $alertMessage
+        ]);
+
+    }
+
+    public function getCategoryById(int $category_id): array|null
+    {
+        if($category_id){
```

Comment:
use ternary operator as much as you can instead of if else when ever there are only one line inside the brackets {}

Reply:
Done